form/auth.go

```diff
@@ -20,22 +20,23 @@ type AuthForm struct {
 	Type         string `json:"type"`
 	SigninMethod string `json:"signinMethod"`
 
-	Organization   string `json:"organization"`
-	Username       string `json:"username"`
-	Password       string `json:"password"`
-	Name           string `json:"name"`
-	FirstName      string `json:"firstName"`
-	LastName       string `json:"lastName"`
-	Gender         string `json:"gender"`
-	Bio            string `json:"bio"`
-	Tag            string `json:"tag"`
-	Education      string `json:"education"`
-	Email          string `json:"email"`
-	Phone          string `json:"phone"`
-	Affiliation    string `json:"affiliation"`
-	IdCard         string `json:"idCard"`
-	Region         string `json:"region"`
-	InvitationCode string `json:"invitationCode"`
+	Organization   string   `json:"organization"`
+	Username       string   `json:"username"`
+	Password       string   `json:"password"`
+	Name           string   `json:"name"`
+	FirstName      string   `json:"firstName"`
+	LastName       string   `json:"lastName"`
+	Gender         string   `json:"gender"`
+	Bio            string   `json:"bio"`
+	Tag            string   `json:"tag"`
+	Education      string   `json:"education"`
+	Email          string   `json:"email"`
+	Phone          string   `json:"phone"`
+	Affiliation    string   `json:"affiliation"`
+	IdCard         string   `json:"idCard"`
+	Region         string   `json:"region"`
+	InvitationCode string   `json:"invitationCode"`
+	Groups         []string `json:"groups"`
```

Comment:
I think we should check whether the group is exist. and according the code, it seems that user can specify group by themselves, will this cause some potential priviledge issue?

Reply:
I could add a check for the existence of the groups but you are right it could cause some privilege issue as I think now. 

Feel like there should be a field in application edit page to select default groups for new users.

Reply:
For now I ended up doing something like this on my side,

```
   ...
    let { tokens, claims } = await this.openIDClientService.callback(
      `${this.appConfig.openid.redirectUri}?code=${code}&state=${res.sessionId}`
    );

    if ((claims as any).groups.length === 0) { //<-- update the user if there's no group
      const user = await this.casdoorService.getUserByUserId(claims.id);
      await this.casdoorService.updateUser({
        ...user,
        groups: this.dynamicConfig.defaultGroups,
      });

      // Refresh the tokenSet
      tokens = await this.openIDClientService.getAccessTokenFromRefreshToken(tokens.refresh_token);
    }
    ...
```